src/python/WMCore/Services/Requests.py

```diff
@@ -255,6 +255,13 @@ def makeRequest_httplib(self, uri=None, data={}, verb='GET',
                 # socket/httplib really screwed up - nuclear option
                 conn.connections = {}
                 raise socket.error('Error contacting: %s: %s' % (self.getDomainName(), msg))
+        except Exception as ex:
+            if "Unable to find the server at" in str(ex):
```

Comment:
I wanted to make it transparent to clients using pycurl instead of httplib2. But I think we can have the same logic as in here:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/Services/Service.py#L62

I'm updating the code soon.